pkg/cloud/api/path.go

```diff
@@ -71,6 +78,26 @@ func (p Path) Equal(other Path) bool {
 	return true
 }
 
+// Similar returns true if other is the same path.
+// For arrays this function does not check indices.
+func (p Path) Similar(other Path) bool {
```

Comment:
I would prefer to call this Match(). Similar() in my mind is a bit ambiguous what it means.